client/my-sites/checkout/cart/cart-item.jsx

```diff
@@ -243,6 +243,10 @@ export class CartItem extends React.Component {
 			name += ' - ' + translate( 'never expires' );
 		}
 
+		if ( cartItem?.extra?.premium ) {
+			name = translate( 'Premium' ) + ' ' + name;
```

Comment:
Should we ensure that we are dealing with domain products before we append this property?